app/lib/dns-server.ts

```diff
@@ -0,0 +1,239 @@
+): Promise<string> => {
+  try {
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "UPSERT",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error(`Could not update the record`);
+    } else {
+      return response.ChangeInfo?.Id;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while updating resource record: ${error}`
+    );
+  }
+};
+
+export const deleteRecord = async (
+  type: RecordType,
+  name: string,
+  value: string,
+  hostedZoneId: string
+): Promise<string> => {
+  try {
+    const command = new ChangeResourceRecordSetsCommand({
+      ChangeBatch: {
+        Changes: [
+          {
+            Action: "DELETE",
+            ResourceRecordSet: {
+              Name: name,
+              Type: type,
+              ResourceRecords: [
+                {
+                  Value: value,
+                },
+              ],
+            },
+          },
+        ],
+      },
+      HostedZoneId: hostedZoneId,
+    });
+
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Id) {
+      throw new Error("Could not delete the record");
+    } else {
+      return response.ChangeInfo?.Id;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while deleting resource record: ${error}`
+    );
+  }
+};
+
+export const listRecords = async (
+  hostedZoneId: string
+): Promise<ResourceRecordSet[]> => {
+  try {
+    const command = new ListResourceRecordSetsCommand({
+      HostedZoneId: hostedZoneId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ResourceRecordSets) {
+      throw new Error("Could not get resource records");
+    } else {
+      return response.ResourceRecordSets;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while fetching resource record: ${error}`
+    );
+  }
+};
+
+export const getChangeStatus = async (requestId: string) => {
+  try {
+    const command = new GetChangeCommand({
+      Id: requestId,
+    });
+    const response = await route53Client.send(command);
+
+    if (!response.ChangeInfo?.Status) {
+      throw new Error("Could not get change info for requested ID");
+    } else {
+      return response.ChangeInfo?.Status;
+    }
+  } catch (error) {
+    throw new Error(
+      `Something went wrong while getting change status: ${error}`
+    );
+  }
+};
+
+export enum RecordType {
+  A = "A",
+  AAAA = "AAAA",
+  CNAME = "CNAME",
+  TXT = "TXT",
+}
+
+export type HostedZone = {
```

Comment:
I don't think you're using this, are you?

Reply:
No, I don't think it's necessary.